v22.2/trigram-indexes.md

```diff
@@ -0,0 +1,354 @@
+
+- The spelling of a search term is not exact.
+- The exact search term is not known.
+
+For example, if you don't know how to spell a name in your database, you can use a [`%` comparison](#comparisons) to perform a fuzzy search. When applied to a `STRING` column, the `%` operator matches values that meet a configured similarity threshold. 
+
+To search for names like "Steven" in column `first_name`:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT first_name FROM users WHERE first_name % 'steven';
+~~~
+
+~~~
+  first_name    
+--------------
+  Stephen
+  Steve
+  Seven
+~~~
+
+{{site.data.alerts.callout_info}}
+Trigram matching only works if the search term has at least 3 characters.
+{{site.data.alerts.end}}
+
+Fuzzy string matching, as well as `LIKE` and `ILIKE` pattern matching, can be very slow on large datasets. To make both types of searches more efficient, create a trigram index. For an example, see [Use a trigram index to speed up fuzzy string matching](#use-a-trigram-index-to-speed-up-fuzzy-string-matching).
+
+### Creation
+
+To create a trigram index, use the [`CREATE INDEX`](create-index.html) syntax that defines an [inverted index](inverted-indexes.html), specifying a `STRING` column and the `gin_trgm_ops` or `gist_trgm_ops` opclass. It is necessary to specify an opclass in order to enable [trigram-based comparisons](#comparisons).
+
+- Using the PostgreSQL-compatible syntax:
+
+  ~~~ sql
+  CREATE INDEX {optional name} ON {table} USING GIN({column} gin_trgm_ops);
+  ~~~
+
+  ~~~ sql
+  CREATE INDEX {optional name} ON {table} USING GIST({column} gist_trgm_ops);
+  ~~~
+
+  {{site.data.alerts.callout_info}}
+  GIN and GiST indexes are implemented identically on CockroachDB. `GIN` and `GIST` are therefore synonymous when defining a trigram index.
+  {{site.data.alerts.end}}
+
+- Using `CREATE INVERTED INDEX`:
+
+  ~~~ sql
+  CREATE INVERTED INDEX {optional name} ON {table} ({column} {opclass});
+  ~~~
+
+### Comparisons
+
+Trigram indexes on `STRING` columns support the following comparison operators:
+
+- **equality**: [`=`](functions-and-operators.html#operators). Note that standard [`btree` secondary indexes](create-index.html#parameters) may perform better than trigram indexes for equality searches.
+- **pattern matching (case-sensitive)**: [`LIKE`](functions-and-operators.html#operators)
+- **pattern matching (case-insensitive)**: [`ILIKE`](functions-and-operators.html#operators)
+- **similarity matching**: [`%`](functions-and-operators.html#operators). This operator returns `true` if the strings in the comparison have a similarity that meets or exceeds the threshold set by the `pg_trgm.similarity_threshold` [session variable](show-vars.html#pg_trgm_similarity_threshold).
+
+For usage examples, see [Use a trigram index to speed up fuzzy string matching](#use-a-trigram-index-to-speed-up-fuzzy-string-matching).
+
+## Examples
+
+### Simple examples
```

Comment:
In my opinion, we might want to stay away from using the word "Simple" especially for examples... as some users may not find it easy. Perhaps the title for this subsection could also be more descriptive like the second one? "Create various trigram indexes" "Create indexes with different trigram matching"... Not sure, if they are helpful 😬 But, some ideas!

Reply:
Agree!